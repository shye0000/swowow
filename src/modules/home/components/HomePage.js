import React from 'react';
import {fetchRoot} from '../redux/actions';
import {connect} from 'react-redux';
import {speedUp} from '../../../redux/actions';
import ResourcesButtons from './ResourcesButtons';
import Animate from 'rc-animate';
import './HomePage.scss';

const Msg = ({fetching, fetchSuccess, counter}) => {

	if (fetching) {
		return <Animate
			transitionName="fade"
			transitionAppear
		>
			<div>Here we go!</div>
		</Animate>;
	}
	if (fetchSuccess) {
		return <Animate
			transitionName="fade"
			transitionAppear
		>
			<div>STAR WARS</div>
		</Animate>;
	}
	if (counter) {
		return null;
	}
	return <Animate
		transitionName="fade"
		transitionAppear
	>
		<div>Bad luck, something went wrong...</div>
	</Animate>;
};

export class HomePage extends React.Component {

	state = {
		counter: 5
	}

	countDownToFetchRoot = () => {
		const count = setInterval(() => {
			this.setState((oldState) => {
				const {counter} = oldState;
				return {counter: counter - 1};
			}, () => {
				const {counter} = this.state;
				if (counter === 0) {
					this.props.fetchRoot();
					clearInterval(count);
				}
			});
		}, 1000);
	}

	componentDidMount = () => {
		const {root, fetchSuccess} = this.props;
		if (!root || !fetchSuccess) {
			this.countDownToFetchRoot();
		}
	}

	render() {
		const {counter} = this.state;
		const {fetching, fetchSuccess, root} = this.props;
		return (
			<div className="home">
				{!fetchSuccess && counter ? <div className="home-counter">{counter}</div> : null}
				<div className="home-msg">
					<Msg fetching={fetching} fetchSuccess={fetchSuccess} counter={counter} />
					<Animate
						transitionName="fade"
						transitionAppear
					>
						{fetchSuccess ? <ResourcesButtons root={root} /> : null}
					</Animate>
				</div>
			</div>
		);
	}
}

const mapStateToProps = (state) => {
	return {
		fetching: state.root.fetching,
		fetchSuccess: state.root.fetchSuccess,
		root: state.root.root
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		fetchRoot: () => {
			dispatch(speedUp());
			dispatch(fetchRoot());
		}
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
