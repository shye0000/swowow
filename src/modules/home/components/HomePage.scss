@import "../../../styles/vars";
.home {
  color: white;
  height: 100%;
  position: relative;
  .home-counter, .home-msg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    font-size: 6em;
    line-height: normal;
    font-weight: bold;
    text-align: center;
    width: 100%;
    padding: $sm-padding $lg-padding;
    max-width: 800px;
  }
  .resources-buttons {
    margin-top: 30px;
    font-size: 0.5em;
    button, a {
      text-transform: uppercase;
      font-weight: bold;
    }
  }
}