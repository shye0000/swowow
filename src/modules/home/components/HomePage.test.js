import React from 'react';
import { HomePage } from './HomePage';
import { shallow } from 'enzyme';

describe('HomePage', () => {

	it.skip('render correctly', () => {
		const props = {
			fetching: false,
			fetchSuccess: false,
			root: null
		};
		const wrapper = shallow(<HomePage {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it.skip('when fetchSuccess "false" or has no root resources in props, call method "countDownToFetchRoot" ', () => {
		const props = {
			fetching: false,
			fetchSuccess: true,
			root: null
		};
		const wrapper = shallow(<HomePage {...props} />);
		const componentDidMountSpy = jest.spyOn(wrapper.instance(), 'componentDidMount');
		const countDownToFetchRootSpy = jest.spyOn(wrapper.instance(), 'countDownToFetchRoot');
		componentDidMountSpy();
		expect(countDownToFetchRootSpy).toBeCalled();
	});

});