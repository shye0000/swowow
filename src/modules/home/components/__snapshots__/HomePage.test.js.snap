// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    fetchSuccess={false}
    fetching={false}
    root={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="home-counter"
        >
          5
        </div>,
        <div
          className="home-msg"
        >
          <Msg
            counter={5}
            fetchSuccess={false}
            fetching={false}
          />
          <Animate
            animation={Object {}}
            component="span"
            componentProps={Object {}}
            onAppear={[Function]}
            onEnd={[Function]}
            onEnter={[Function]}
            onLeave={[Function]}
            transitionAppear={true}
            transitionEnter={true}
            transitionLeave={true}
            transitionName="fade"
          />
        </div>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 5,
          "className": "home-counter",
        },
        "ref": null,
        "rendered": 5,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Msg
              counter={5}
              fetchSuccess={false}
              fetching={false}
            />,
            <Animate
              animation={Object {}}
              component="span"
              componentProps={Object {}}
              onAppear={[Function]}
              onEnd={[Function]}
              onEnter={[Function]}
              onLeave={[Function]}
              transitionAppear={true}
              transitionEnter={true}
              transitionLeave={true}
              transitionName="fade"
            />,
          ],
          "className": "home-msg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "counter": 5,
              "fetchSuccess": false,
              "fetching": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animation": Object {},
              "children": null,
              "component": "span",
              "componentProps": Object {},
              "onAppear": [Function],
              "onEnd": [Function],
              "onEnter": [Function],
              "onLeave": [Function],
              "transitionAppear": true,
              "transitionEnter": true,
              "transitionLeave": true,
              "transitionName": "fade",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="home-counter"
          >
            5
          </div>,
          <div
            className="home-msg"
          >
            <Msg
              counter={5}
              fetchSuccess={false}
              fetching={false}
            />
            <Animate
              animation={Object {}}
              component="span"
              componentProps={Object {}}
              onAppear={[Function]}
              onEnd={[Function]}
              onEnter={[Function]}
              onLeave={[Function]}
              transitionAppear={true}
              transitionEnter={true}
              transitionLeave={true}
              transitionName="fade"
            />
          </div>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "className": "home-counter",
          },
          "ref": null,
          "rendered": 5,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Msg
                counter={5}
                fetchSuccess={false}
                fetching={false}
              />,
              <Animate
                animation={Object {}}
                component="span"
                componentProps={Object {}}
                onAppear={[Function]}
                onEnd={[Function]}
                onEnter={[Function]}
                onLeave={[Function]}
                transitionAppear={true}
                transitionEnter={true}
                transitionLeave={true}
                transitionName="fade"
              />,
            ],
            "className": "home-msg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "counter": 5,
                "fetchSuccess": false,
                "fetching": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animation": Object {},
                "children": null,
                "component": "span",
                "componentProps": Object {},
                "onAppear": [Function],
                "onEnd": [Function],
                "onEnter": [Function],
                "onLeave": [Function],
                "transitionAppear": true,
                "transitionEnter": true,
                "transitionLeave": true,
                "transitionName": "fade",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
